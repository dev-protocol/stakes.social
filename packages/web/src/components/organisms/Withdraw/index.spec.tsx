import React from 'react'
import { message } from 'antd'
import { act, render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { getMyStakingAmount } from 'src/fixtures/dev-kit/client'
import { Withdraw } from '.'
import 'src/__mocks__/window/matchMedia.mock'
import WalletContext from 'src/context/walletContext'
import { ethers } from 'ethers'

jest.mock('src/fixtures/dev-kit/client')
jest.mock('src/fixtures/dev-kit/hooks')
jest.mock('src/fixtures/uniswap/hooks')
jest.mock('src/fixtures/wallet/hooks')
jest.mock('@dev/graphql')

describe(`${Withdraw.name}`, () => {
  test('Snapshot', () => {
    const component = render(
      <WalletContext.Provider value={{ ethersProvider: ethers.getDefaultProvider(), setProviders: () => {} }}>
        <Withdraw propertyAddress="propertyAddress" />
      </WalletContext.Provider>
    )
    const tree = component.baseElement
    expect(tree).toMatchSnapshot()
  })
  test('hooks: click MAX button and withdraw button', () => {
    ;(getMyStakingAmount as jest.Mock).mockImplementation(() => Promise.resolve({ amount: 1000 }))
    message.warn = jest.fn(() => {}) as any

    const { getByText } = render(
      <WalletContext.Provider value={{ ethersProvider: ethers.getDefaultProvider(), setProviders: () => {} }}>
        <Withdraw propertyAddress="propertyAddress" />
      </WalletContext.Provider>
    )

    act(() => {
      userEvent.click(getByText('DEV'))
      userEvent.click(getByText('Withdraw'))
    })

    // expect((message.warn as jest.Mock).mock.calls.length).toBe(0)
    expect((getMyStakingAmount as jest.Mock).mock.calls.length).toBe(1)
  })
})
