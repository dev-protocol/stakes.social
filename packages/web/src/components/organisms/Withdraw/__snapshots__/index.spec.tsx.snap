// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Withdraw Snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-gKclnd gwKHOb"
    >
      <div
        class="sc-hKwDye gWsDYA"
        style="opacity: 1;"
      >
        <div
          class="ant-radio-group ant-radio-group-outline"
          style="margin-bottom: 12px;"
        >
          <div
            class="ant-space ant-space-vertical"
          >
            <div
              class="ant-space-item"
              style="margin-bottom: 8px;"
            >
              <label
                class="ant-radio-wrapper"
              >
                <span
                  class="ant-radio"
                >
                  <input
                    class="ant-radio-input"
                    type="radio"
                    value="2"
                  />
                  <span
                    class="ant-radio-inner"
                  />
                </span>
                <span>
                  <div
                    class="ant-space ant-space-horizontal ant-space-align-center"
                  >
                    <div
                      class="ant-space-item"
                      style="margin-right: 24px;"
                    >
                      <div
                        style="margin-left: 8px; min-width: 40px;"
                      >
                        #
                        2
                      </div>
                    </div>
                    <div
                      class="ant-space-item"
                      style="margin-right: 24px;"
                    >
                      <div
                        style="min-width: 120px;"
                      >
                        <span>
                          1e-15
                        </span>
                        <span
                          style="font-size: 0.6em; margin-left: 2px;"
                        >
                          Staked
                        </span>
                      </div>
                    </div>
                    <div
                      class="ant-space-item"
                    >
                      <div>
                        <span>
                          0
                        </span>
                        <span
                          style="font-size: 0.6em; margin-left: 2px;"
                        >
                          Claimable
                        </span>
                      </div>
                    </div>
                  </div>
                </span>
              </label>
            </div>
            <div
              class="ant-space-item"
              style="margin-bottom: 8px;"
            >
              <label
                class="ant-radio-wrapper"
              >
                <span
                  class="ant-radio"
                >
                  <input
                    class="ant-radio-input"
                    type="radio"
                    value="4"
                  />
                  <span
                    class="ant-radio-inner"
                  />
                </span>
                <span>
                  <div
                    class="ant-space ant-space-horizontal ant-space-align-center"
                  >
                    <div
                      class="ant-space-item"
                      style="margin-right: 24px;"
                    >
                      <div
                        style="margin-left: 8px; min-width: 40px;"
                      >
                        #
                        4
                      </div>
                    </div>
                    <div
                      class="ant-space-item"
                      style="margin-right: 24px;"
                    >
                      <div
                        style="min-width: 120px;"
                      >
                        <span>
                          1e-15
                        </span>
                        <span
                          style="font-size: 0.6em; margin-left: 2px;"
                        >
                          Staked
                        </span>
                      </div>
                    </div>
                    <div
                      class="ant-space-item"
                    >
                      <div>
                        <span>
                          0
                        </span>
                        <span
                          style="font-size: 0.6em; margin-left: 2px;"
                        >
                          Claimable
                        </span>
                      </div>
                    </div>
                  </div>
                </span>
              </label>
            </div>
            <div
              class="ant-space-item"
            >
              <label
                class="ant-radio-wrapper"
              >
                <span
                  class="ant-radio"
                >
                  <input
                    class="ant-radio-input"
                    type="radio"
                    value="6"
                  />
                  <span
                    class="ant-radio-inner"
                  />
                </span>
                <span>
                  <div
                    class="ant-space ant-space-horizontal ant-space-align-center"
                  >
                    <div
                      class="ant-space-item"
                      style="margin-right: 24px;"
                    >
                      <div
                        style="margin-left: 8px; min-width: 40px;"
                      >
                        #
                        6
                      </div>
                    </div>
                    <div
                      class="ant-space-item"
                      style="margin-right: 24px;"
                    >
                      <div
                        style="min-width: 120px;"
                      >
                        <span>
                          1e-15
                        </span>
                        <span
                          style="font-size: 0.6em; margin-left: 2px;"
                        >
                          Staked
                        </span>
                      </div>
                    </div>
                    <div
                      class="ant-space-item"
                    >
                      <div>
                        <span>
                          0
                        </span>
                        <span
                          style="font-size: 0.6em; margin-left: 2px;"
                        >
                          Claimable
                        </span>
                      </div>
                    </div>
                  </div>
                </span>
              </label>
            </div>
          </div>
        </div>
        <span
          class="ant-input-group-wrapper ant-input-group-wrapper-lg ant-input-search ant-input-search-large ant-input-search-with-button sc-dkPtRN jgEzXX"
        >
          <span
            class="ant-input-wrapper ant-input-group"
          >
            <span
              class="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
            >
              <input
                class="ant-input ant-input-lg"
                id="withdraw"
                type="number"
                value="0"
              />
              <span
                class="ant-input-suffix"
              >
                <button
                  class="sc-bdvvtL bxKQgB"
                >
                  DEV
                </button>
              </span>
            </span>
            <span
              class="ant-input-group-addon"
            >
              <button
                class="ant-btn ant-btn-primary ant-btn-lg ant-input-search-button"
                type="button"
              >
                <span>
                  Withdraw
                </span>
              </button>
            </span>
          </span>
        </span>
      </div>
      <div
        class="sc-furwcr deVbcw"
      >
        <div
          class="sc-iCfMLu jZywgv"
        >
          <span
            aria-label="info-circle"
            class="anticon anticon-info-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="info-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
              <path
                d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
              />
            </svg>
          </span>
        </div>
        <span
          style="font-size: 0.8em;"
        >
          You will receive all accumulated rewards when withdrawing any amount of staked DEV.
        </span>
      </div>
    </div>
  </div>
</body>
`;
